kip-0017.md

```diff
@@ -8,105 +8,426 @@ Category: Interface
+
+The method response contains a `result` object that contains the following properties:
+
+1. `result`: `Object`  
+    1.1. `accounts`: `Array<Object>`  
+    __ 1.1.1 `account`: `string` - the Wallet Connect [Account](#wallet-connect-accounts) requested.  
+    __ 1.1.2 `publicKey`: `string` - the public key extracted from the Wallet Connect account requested.  
+    __ 1.1.3 `kadenaAccounts`: `Array<Object>`   
+    ____ 1.1.3.1 `name`: `string` - the account name as stored in the blockchain for the specified contract.  
+    ____ 1.1.3.2 `contract`: `string` - the contract that this account is used for.  
+    ____ 1.1.3.3 `chains`: `Array<string>` - the chains on which this account is present.  
+
+> **NB**: The accounts that are returned are the ones that are available in the 
+> scope of the wallet (i.e. only the accounts the wallet knows about).
+
+> **NB**: The same Kadena account name can be used in different contracts 
+> (e.g. `coin`, `my-fungible`). They require different entries 
+> since the chains on which the accounts are present can differ.
+
+### kadena_quickSign_v1
+
+#### Example kadena_quickSign_v1 Request and Response
+
+```jsonc
+// Signing Request
 {
-  "kadena": {
-    "accounts": [
-      "kadena:mainnet01:38298612cc2d5e841a232bd08413aa5304f9ef3251575ee182345abc3807dd89",
-      "kadena:testnet04:38298612cc2d5e841a232bd08413aa5304f9ef3251575ee182345abc3807dd89",
-      "kadena:testnet04:22ddc64851718e9d41d98b0f33d5e328ae5bbbbd97aed9885adac0f2d070ff9c"
-    ],
-    "methods": ["kadena_quicksign_v1"],
-    "events": []
+  "commandSigDatas": [CommandSigData]
+}
+```
+
+The structure of `CommandSigData` is defined in the
+[kip-0015 quicksign signing api v1](https://github.com/kadena-io/KIPs/blob/master/kip-0015.md#commandsigdata)
+
+```jsonc
+// Succesfull Signing Response
+{
+  "responses": [Response]
+}
+```
+
+or
+
+```jsonc
+// Failed Signing Response
+{
+  "error": QuicksignError
+}
+
+```
+
+The structure of `Response` and `QuicksignError` are defined in the
+[kip-0015 quicksign signing api v1](https://github.com/kadena-io/KIPs/blob/master/kip-0015.md#quicksign-response)
+
+### kadena_sign_v1
+
+#### Example kadena_sign_v1 Request and Response
+
+```jsonc
+// Signing request
+{
+  signingCmd
+}
+
+```
+
+The structure of `signingCmd` is defined in the
+[Chainweaver Signing Api](https://github.com/kadena-io/pact-lang-api#chainweaver-signing-api-command)
+
+```jsonc
+# Signing response
+{
+  "signedCmd": {
+      cmd: ...,
+      hash: ...
+      sigs: [
+        ...
+      ]
     }
 }
 ```
 
-# ****Backwards Compatibility****
+## WalletConnect Events
+
+At the moment we see no need to standardize any events and believe that most
+use cases can be handled at the Kadena protocol level that sits atop Wallet
+Connect.
+
+This is subject to change in the future as we understand the workflows more through extended use of the protocol.
+
+However, any future Kadena event names will formatted as:
```

Comment:
Thought having this section would be useful to make sure any kadena- or community-driven events follow a specific standard.

Reply:
I also added a similar section in the `WalletConnect Methods` section.
Not sure about the wording of the NB section.